### DISCUSSION TITLE
'additionalProperties' is required to be supplied and to be false.

### DISCUSSION THREAD

**Discussion by MabudAlam** [outlines-dev/outlines#1219] [Q&A]
Hi , 
I am using the openai gpt 4 o mini with pydantic model . But I keep getting this error.

**{
    "detail": "500: Error code: 400 - {'error': {'message': \"Invalid schema for response_format 'default': In context=(), 'additionalProperties' is required to be supplied and to be false.\", 'type': 'invalid_request_error', 'param': 'response_format', 'code': None}}"
}**

### My Code :

 model = models.openai(
            "gpt-4o-mini",
            api_key=get_env_variable("OpenAI-Key-v2"),
        )
        generator = generate.json(model, SkillInto)
        prompt = f"""
        Create a skill template for a new skill with the following description:
        {query.skillDescription}
        
        Please provide the following details:
        - skillName
        - skillDescription
        - category (must be one of: Sports, Music, Other)
        """

        result = generator(prompt)
from enum import Enum

from pydantic import BaseModel, conint, constr

class SkillCategory(str, Enum):
    sports = "Sports"
    music = "Music"
    other = "Other"

class SkillInto(BaseModel):
    skillName: str
    skillDesc: str
    category: SkillCategory

**Comment by lapp0** [comment]
Please try setting the `model_config` attribute as shown here :

https://github.com/openai/openai-python/blob/e1b2f8216cc69e802475a0d438e40e0e74510de4/src/openai/cli/_models.py#L12C9-L12C21

**Comment by arpadtamasi** [comment]
In my case it was caused by an accidental

     from openai import OpenAI, BaseModel

worked like a charm after changing to

     from openai import OpenAI
     from pydantic import BaseModel

**Reply by dlvovsky** [reply]
Thanks!

**Comment by sachin-som** [comment]
Adding `extra = 'forbid'` in the Pydantic `Config` class works for me.

You can try this -

```
class SkillInto(BaseModel):
    skillName: str
    skillDesc: str
    category: SkillCategory

    class Config:
        extra = "forbid"
```

**Reply by FNDEVVE** [reply]
GOAT

**Reply by mimoo** [reply]
wait that can't be the answer :D

**Comment by RobinPicard** [comment]
Closing the discussion as we have solved the issue by setting `additionalProperties` to False by default in the `OpenAI` model.
